### PR TITLE
[ACS-6190] prefer-promise-reject-errors rule and fixes

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- fix the SonarCloud issues for promise rejections
- enable corresponding eslint rule to prevent issues in the future

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://alfresco.atlassian.net/browse/ACS-6190

### CODE REVIEW

File: lib/content-services/src/lib/common/services/rendition.service.ts [hunk 1]
```diff
@@ -201,7 +201,7 @@ export class RenditionService {
                                 clearInterval(intervalId);
                                 return resolve(this.handleNodeRendition(nodeId, rendition.entry.content.mimeType, versionId));
                             }
-                        }, () => reject());
+                        }, () => reject(new Error('error')));
```
Comment (MichalKinas) [inline lib/content-services/src/lib/common/services/rendition.service.ts:204]: Should we provide something more meaningful as a error message here?

File: lib/content-services/src/lib/common/services/rendition.service.ts [hunk 2]
```diff
@@ -210,11 +210,11 @@ export class RenditionService {
                                 clearInterval(intervalId);
                                 return resolve(this.handleNodeRendition(nodeId, renditionId, versionId));
                             }
-                        }, () => reject());
+                        }, () => reject(new Error('error')));
```
Comment (MichalKinas) [inline lib/content-services/src/lib/common/services/rendition.service.ts:213]: Same here

File: lib/content-services/src/lib/common/services/rendition.service.ts [hunk 3]
```diff
@@ -210,11 +210,11 @@ export class RenditionService {
...
                     }
                 } else {
                     clearInterval(intervalId);
-                    return reject();
+                    return reject(new Error('error'));
```
Comment (MichalKinas) [inline lib/content-services/src/lib/common/services/rendition.service.ts:217]: Same here

File: lib/process-services/src/lib/process-list/services/process-filter.service.ts [hunk 4]
```diff
@@ -87,8 +85,7 @@ export class ProcessFilterService {
     getProcessFilterByName(filterName: string, appId?: number): Observable<FilterProcessRepresentationModel> {
         return from(this.callApiProcessFilters(appId))
             .pipe(
-                map((response: any) => response.data.find((filter) => filter.name === filterName)),
-                catchError((err) => this.handleProcessError(err))
+                map((response: any) => response.data.find((filter) => filter.name === filterName))
```
Comment (MichalKinas) [inline lib/process-services/src/lib/process-list/services/process-filter.service.ts:88]: We can remove `any` here right?

File: lib/process-services/src/lib/process-list/services/process-filter.service.ts [hunk 5]
```diff
@@ -72,8 +71,7 @@ export class ProcessFilterService {
     getProcessFilterById(filterId: number, appId?: number): Observable<FilterProcessRepresentationModel> {
         return from(this.callApiProcessFilters(appId))
             .pipe(
-                map((response: any) => response.data.find((filter) => filter.id === filterId)),
-                catchError((err) => this.handleProcessError(err))
+                map((response: any) => response.data.find((filter) => filter.id === filterId))
```
Comment (MichalKinas) [inline lib/process-services/src/lib/process-list/services/process-filter.service.ts:74]: We can remove `any` here right?

File: lib/content-services/src/lib/common/services/rendition.service.ts (hunk 3 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/common/services/rendition.service.ts]: fixed

File: lib/content-services/src/lib/common/services/rendition.service.ts (hunk 2 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/common/services/rendition.service.ts]: fixed

File: lib/content-services/src/lib/common/services/rendition.service.ts (hunk 1 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/common/services/rendition.service.ts]: added some messages

File: lib/process-services/src/lib/process-list/services/process-filter.service.ts (hunk 4 above)
Comment (DenysVuika) [reply lib/process-services/src/lib/process-list/services/process-filter.service.ts]: removed and fixed the issue with incorrect return type :)

File: lib/process-services/src/lib/process-list/services/process-filter.service.ts (hunk 5 above)
Comment (DenysVuika) [reply lib/process-services/src/lib/process-list/services/process-filter.service.ts]: removed and fixed typing issue discovered after that :)

**Review by eromano — APPROVED** [review]

**Pull request merged**